### ISSUE TITLE
Please implement better searching

### ISSUE THREAD

**Issue by AAHecv** [MicrosoftDocs/feedback#2410]
**Is your feature request related to a problem? Please describe.**
I hate to be someone who longs for the good old days, but I miss the TechNet site.  I used it almost daily, and it was the first place I would look no matter the technical issue.  The searching was so much better, and gave the ability to narrow by topic.  While the solution might still exist in the new site, it returns so many (seemingly) unrelated hits, it is all but useless.  
**Describe the solution you'd like**
I would love to be able to search by platform (i.e. Server 2019) and keywords within the search criteria, such as Event ID 1234 and Server 2019.
**Describe alternatives you've considered**
Until now, TechNet and Microsoft were my first stops, now I am wasting time using Google and trying to discern legitimate sites.
**Additional context**
I would be glad to be a beta tester on any new solutions / replacements.  Thank you for your consideration.  I am not asking someone to re-invent the wheel, but I sure would like access to the wheels that already existed.

**Comment by oising** [comment]
Agreed. The search results are essentially random. Quoting words as a phrase (e.g. "bad request" matches bad and request independently) doesn't work at all. The search is painfully slow. Sometimes search leads to the page stopping, or returning to home. Other times it fails with a "too many redirects" error. It's completely broken.

**Comment by greyseal96** [comment]
I hope this isn't necrobumping but I wanted add on an example to this.  

I need to write a small application to do some stuff in C# with IIS.  I found by reading online that I needed to use the Microsoft.Web.Administration namespace so I went to docs.microsoft.com and entered "microsoft.web.administration" into the search box and, surprisingly, I got no results.  I don't mean some results that weren't really related or not useful or anything like that.  I mean literally "No results; Sorry, we couldn't find anything that fit your search" Here's the link to the search in case that helps:  https://docs.microsoft.com/en-us/search/?terms=microsoft.web.administration  That was pretty surprising.

I went to my favorite search engine and did a quick search for the same thing (microsoft.web.administration) and the first result I got was to a page on the Docs site.  So the non-Docs site search engine found it but not the Docs site's own search engine :-/ .  Here's the link to the page:  https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.administration?view=iis-dotnet

In my mind, the minimum expectation would be that I at least get a result to the page listed above for the namespace.  A "nice to have" would be to also return related results like any other Docs pages which deal with that namespace like tutorials, walkthroughs, etc.
